lib/puppet/pops/evaluator/deferred_resolver.rb

```diff
@@ -14,13 +14,12 @@ class DeferredResolver
   # Deferred values inside of custom Object instances are not resolved as this
   # is expected to be done by such objects.
   #
-  # @param node [Puppet::Node] the node for the catalog
   # @param facts [Puppet::Node::Facts] the facts object for the node
   # @param catalog [Puppet::Resource::Catalog] the catalog where all deferred values should be replaced
   # @return [nil] does not return anything - the catalog is modified as a side effect
   #
-  def self.resolve_and_replace(node, facts, catalog)
-    compiler = Puppet::Parser::ScriptCompiler.new(node.environment, node.name)
+  def self.resolve_and_replace(facts, catalog)
+    compiler = Puppet::Parser::ScriptCompiler.new(catalog.environment_instance, catalog.name)
```

Comment:
In my testing, node.name and catalog.name where interchangeable; is there a case where they might not be?

Reply:
If the node changed name since it got the catalog?
The change is ok if it is known that catalog.environment_instance is the same thing and catalog.name is the same as node.name